test/python/circuit/test_library.py

```diff
@@ -50,3 +55,108 @@ def test_inner_product(self):
         expected.cz(1, 4)
         expected.cz(2, 5)
         self.assertEqual(circuit, expected)
+
+
+@ddt
+class TestBasisChanges(QiskitTestCase):
+    """Test the basis changes."""
+
+    def to_matrix(self, circuit):
+        """Get the unitary matrix from simulating the circuit with the unitary simulator."""
+        backend = BasicAer.get_backend('unitary_simulator')
+        return execute(circuit, backend).result().get_unitary()
+
+    def assertQFTIsCorrect(self, qft, num_qubits=None, inverse=False, prepend_swaps=False):
+        """Assert that the QFT circuit produces the correct matrix.
+
+        Can be provided with an explicit number of qubits, if None is provided the number
+        of qubits is set to ``qft.num_qubits``.
+        """
+        if prepend_swaps:
+            circuit = QuantumCircuit(*qft.qregs)
+            for i in range(circuit.num_qubits // 2):
+                circuit.swap(i, circuit.num_qubits - i - 1)
+
+            qft = circuit + qft
+
+        simulated = self.to_matrix(qft)
```

Comment:
instead of using basic_aer, you can just do
```
from qiskit.quantum_info import Operator
lhs = Operator(qft)   # simulates a circuit
rhs = Operator(expected)   # converts a matrix to an operator
```
and then an operator has methods for checking exact equality (`lhs == rhs`) and equality up to global phase (`lhs.equiv(rhs)`) and it also allows you to set a `atol` and `rtol` so you don't have to rely on `np.testing.assert_array_almost_equal`